src/Zenith.sol

```diff
@@ -27,9 +30,18 @@ contract Zenith is Passage {
     }
 
     /// @notice The host block number that a block was last submitted at for a given rollup chainId.
-    /// rollupChainId => host blockNumber that block was last submitted at
+    /// rollupChainId => host blockNumber the last block was submitted at.
     mapping(uint256 => uint256) public lastSubmittedAtBlock;
 
+    /// @notice The gasLimit of the last submitted block for a given rollup chainId.
+    /// @dev NOTE that this can change mid-block, causing some `transact` to have a different limit than others.
```

Comment:
starting at this WIP code to figure the best way to fix this (or if it's necessary)

Reply:
consider scenario: 
- previous block gasLimit is 30M
- 28M transact gas is used in a block
- current block is then submitted with gasLimit is 25M
- `currentGasLimit` is updated to 25M, while `transactGasUsed` for this block is already 28M

Reply:
that's not necessarily the case, you'd just have to store the last block's limit the first time a transact occurs. bears more discussion tho